internal/orchestrationapi/orchestration.go

```diff
@@ -201,7 +201,7 @@ func (o OrchestrationBuilder) Build() Orche {
 func (o *orcheImpl) Start(deviceIDPath string, platform string, executionType string) {
 	resourceMonitorImpl.StartMonitoringResource()
 	o.discoverIns.StartDiscovery(deviceIDPath, platform, executionType)
-	o.storageIns.StartStorage()
+	o.storageIns.StartStorage("")
```

Comment:
How is the handling of host being run in different machine taken care, since null is being passed from orchestration, wanted to check on this.

Reply:
Thanks for the review!

The host is the address of edgex-data and edgex-metadata, and DataStorage of edge-orchestration tries to connect to edgex with that address.
DataStorage in a edge-orchestration should not handle or control other edge-orchestrations.

Let me explain briefly about StartStorage() with the host, which is set to "".
1. StartStorage checks whether SERVICE environment variable contains the word, DataStorage.
2-1. If edge-orchestration has the environment variable, it generates configuration files with host "127.0.0.1".
2-2. If edge-orchestration doesn't have the environment variable, it checks the host address and generates configuration files when the host is not "".
3. Run DataStorage server with the configuration generated.
* If edge-orchestration doesn't have environment variable in terms of DataStorage and doesn't have proper host address, the edge-orchestration runs with the configuration files in the /var/edge-orchestration/datastorage/.

I'm not sure if I have answered your question sufficiently.
Please refer to the source code of StartStorage() for more information.

Reply:
Thanks for your kind reply, after referring to files had posted the query. Anyways thanks for the details.